File: 250429/그 요일은/the-day-of-the-day.cpp
#include <iostream>
using namespace std;

int Days(int month, int day) {
    int days[13] = {0, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
    int total_days = 0;
    for (int i = 1; i < month; i++) {
        total_days += days[i];
    }
    total_days += day;

    return total_days;
}

int Weekday(string s) {
    if(s == "Mon") return 0;
    else if (s == "Tue") return 1;
    else if(s == "Wed") return 2;
    else if(s == "Thu") return 3;
    else if(s == "Fri") return 4;
    else if(s == "Sat") return 5;
    else if(s == "Sun") return 6;
}
int main() {
    ios::sync_with_stdio(0);
    cin.tie(NULL);
    cout.tie(NULL);

    int m1, d1, m2, d2;
    string A;
    int ans = 0;
    cin >> m1 >> d1 >> m2 >> d2 >> A;
    int start = Days(m1, d1);
    int end = Days(m2, d2);
    int current = Weekday("Mon");

    for(int i = start; i <= end; i++) {
        if(current == Weekday(A)) {
            ans++;
        }
        current = (current + 1) % 7;
    }

    cout << ans;
    return 0;
}

File: 250429/그 요일은/the-day-of-the-day.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        // 입력받기
        int m1 = sc.nextInt(); // 시작 월
        int d1 = sc.nextInt(); // 시작 일
        int m2 = sc.nextInt(); // 목표 월
        int d2 = sc.nextInt(); // 목표 일
        String targetDay = sc.next(); // 찾고자 하는 요일 (예: "Mon")

        // 요일 배열과 월별 일수 배열
        String[] weekdays = new String[] { "Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun" };
        int[] daysInMonth = new int[] { 0, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 }; // 2024년은 윤년

        // 시작 요일의 인덱스
        int startIndex = 0; // 월요일은 0번째 인덱스

        // 날짜 간의 총 일수 계산
        int totalDays = 0;
        if (m1 == m2) {
            totalDays = d2 - d1;
        } else {
            totalDays += daysInMonth[m1] - d1; // 첫 달의 남은 일수
            for (int i = m1 + 1; i < m2; i++) {
                totalDays += daysInMonth[i]; // 중간 달의 일수
            }
            totalDays += d2; // 마지막 달의 일수
        }

        // 요일 등장 횟수 계산
        int targetIndex = 0;
        for (int i = 0; i < weekdays.length; i++) {
            if (weekdays[i].equals(targetDay)) {
                targetIndex = i;
                break;
            }
        }

        int count = 0;
        for (int i = 0; i <= totalDays; i++) {
            int currentDayIndex = (startIndex + i) % 7;
            if (currentDayIndex == targetIndex) {
                count++;
            }
        }

        // 결과 출력
        System.out.println(count);

        sc.close();
    }
}